runners/spark/src/main/java/org/apache/beam/runners/spark/util/CachedSideInputReader.java

```diff
@@ -86,9 +55,27 @@ private CachedSideInputReader(SideInputReader delegate) {
   @Override
   public <T> T get(PCollectionView<T> view, BoundedWindow window) {
```

Comment:
I am worried on the possible semantics consequences of get returning a `null` value when it is not in the `Cache` for this `SideInputReader` wouldn't it imply that a window could get an empty side input assigned?
[The documentation](https://github.com/apache/beam/blob/6c2fe17cfdea1be1fdcfb02267894f0d37a671b3/runners/core-java/src/main/java/org/apache/beam/runners/core/SideInputReader.java#L32) on this is not really clear (pinging @kennknowles to see if I am misreading it).
Wonder if there is a test to validate that this cannot happen or if we can create one somehow?

Reply:
Ah, the meaning of that comment is that `null` is a value. You can have a `PCollection<@Nullable Foo>` that contains just one copy of `null` and use `View.asSingleton()` and the side input returns the `null`.

In other words, `get` _must_ return a value of type `T`. But the type `T` may itself be `@Nullable Something`. The annotation on `SideInputReader` should be removed. It is is incorrect if we use a static analysis that understands this. Findbugs does not understand this but we should aspire for our annotations to be correct so the documentation is clear.